ddtrace/contrib/httplib/patch.py

```diff
@@ -183,13 +184,12 @@ def _wrap_putheader(func, instance, args, kwargs):
 
 def should_skip_request(pin, request):
     """Helper to determine if the provided request should be traced"""
+    if getattr(request, _HTTPLIB_NO_TRACE_REQUEST, False):
+        return True
+
     if not pin or not pin.enabled():
         return True
 
-    agent_url = pin.tracer.agent_trace_url
```

Comment:
Actually I think this logic is used in the RC, instrumentation telemetry, profiling, and di httpclients 😓. Refactoring this could be out of the scope of this PR. We should probably add this logic back and leave a comment.